### PR TITLE
Fix txn limit error when querycoord remove task from etcd

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #12083 
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querycoord/task_scheduler.go [hunk 1]
```diff
@@ -631,21 +631,27 @@ func (scheduler *TaskScheduler) scheduleLoop() {
...
+			childTaskKeys = append(childTaskKeys, taskKey)
+			childTaskKeys = append(childTaskKeys, stateKey)
+			err := scheduler.client.MultiRemove(childTaskKeys)
+			// after recover, child Task's state will be TaskDone, will not be repeat executed
+			if err != nil {
+				panic(err)
+			}
 		}
-		err := scheduler.client.MultiRemove(keys)
+		triggerTaskKeys := make([]string, 0)
```
Comment (congqixia) [inline internal/querycoord/task_scheduler.go:647]: Please use
```
var triggerTaskKeys []string
```

File: internal/querycoord/querynode.go [hunk 2]
```diff
@@ -255,11 +257,12 @@ func (qn *queryNode) releasePartitionsInfo(collectionID UniqueID, partitionIDs [
 			}
 		}
 		info.PartitionIDs = newPartitionIDs
-		err := removeNodeCollectionInfo(collectionID, qn.id, qn.kvClient)
+		err := saveNodeCollectionInfo(collectionID, info, qn.id, qn.kvClient)
```
Comment (congqixia) [inline internal/querycoord/querynode.go:260]: Will there be any side effect after all partitions of a collection is released?

Comment (xige-16) [reply internal/querycoord/querynode.go:260]: wrong use of removeNodeCollectionInfo

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/querycoord/task_scheduler.go (hunk 1 above)
Comment (xige-16) [reply internal/querycoord/task_scheduler.go:647]: will fix in next pr

**Comment by xige-16** [comment]
link #11994

**Comment by xige-16** [comment]
fix #12065

**Pull request merged**